src/connectionManager.ts

```diff
@@ -11,46 +13,68 @@ export interface Connection {
 }
 
 export class ConnectionManager {
+  public static connectionManager: ConnectionManager;
   private connections: Record<string, Connection>;
   private currentConnection?: Connection;
+  private static _extContext: ExtensionContext;
 
-  constructor() {
+  private constructor() {
     this.connections = {};
   }
 
+  public static _initialize(ec: ExtensionContext) {
+    ConnectionManager.connectionManager = new ConnectionManager();
+    ConnectionManager.connectionManager.connections = {};
+    this._extContext = ec;
+
+    if (this._extContext.globalState.get('ConnectionManager.connections') != undefined) {
+      this.connectionManager.connections = this._extContext.globalState.get(
+        'ConnectionManager.connections',
+      ) as Record<string, Connection>;
+    }
+
+    if (this._extContext.globalState.get('ConnectionManager.currentConnection') != undefined) {
```

Comment:
`!=` in JavaScript/TypeScript can lead to subtle type errors; always use [strict equality](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_inequality) instead.
```suggestion
    if (this._extContext.globalState.get('ConnectionManager.currentConnection') !== undefined) {
```